lib/rules/no-constant-condition.js

```diff
@@ -97,6 +97,9 @@ module.exports = {
                 case "ObjectExpression":
                 case "ArrayExpression":
                     return true;
+                case "TemplateLiteral":
+                    return (inBooleanPosition && node.quasis.some(quasi => quasi.value.cooked.length)) ||
```

Comment:
:+1: 

It might be nice to also add a valid test like the following one, to make sure that the `cooked` check doesn't get changed to `raw` or some other way to examine template literal's content:

```js
"if (`\\\n${a}`) {}"
```

It represents this code (linebreak immediately after `\`):

```js
if (`\
${a}`) {}
```

(I'm not sure is there any other sequence that also results in an empty string)

Reply:
@mdjermanovic 
 I added test cases about it. Thanks! :)